### DISCUSSION TITLE
clj-kondo errors hide arities in Emacs

### DISCUSSION THREAD

**Discussion by jcf** [clojure-lsp/clojure-lsp#1911] [Q&A]
When writing code in Emacs, clojure-lsp will helpfully suggest arguments one can pass to the function under point via the minibuffer.

![CleanShot 2024-11-05 at 23 35 31@2x](https://github.com/user-attachments/assets/5005206a-33c9-4854-abe0-0e7e81fe08e2)

Unfortunately, until you have all of the necessary parameters in place, clojure-lsp surfaces clj-kondo linting issues in the Emacs minibuffer like so:

![CleanShot 2024-11-05 at 23 36 47@2x](https://github.com/user-attachments/assets/bf60d906-a20b-4ba7-93ef-00edcb6d1cfb)

One can supress the linting errors by switching `:diagnostics`, but at the cost of slightly less accurate diagnostics reporting:

``` clj
{:diagnostics {:range-type :simple}}
```

Is there a more elegant solution to this problem I've not considered? I'm unsure how clojure-lsp can do anything 'better' in this case. Maybe showing both the arities and linting issues in the minibuffer is possible?

**Comment by ericdallo** [comment]
@jcf I don't think showing the diagnostics in the minibuffer is a good usage of the minibuffer, that's why I use lsp-ui sideline diagnostics feature:
![foo](https://github.com/user-attachments/assets/dc34bedd-220a-40f4-8ef4-cf8baa8aaed9)
